app/gui/view/src/project.rs

```diff
@@ -26,6 +27,45 @@ use ensogl_hardcoded_theme::Theme;
 
 
 
+// ============================
+// === WayOfOpeningSearcher ===
+// ============================
+
+/// An enum describing how the searcher was opened.
+#[derive(Clone, CloneRef, Copy, Debug, PartialEq)]
+pub enum WayOfOpeningSearcher {
+    /// New node was created by opening the searcher or the node is being edited.
+    NodeEdited(NodeId),
+    /// New node was created by dropping a dragged connection on the scene.
+    EdgeDropped(NodeId, EdgeId),
+}
+
+impl WayOfOpeningSearcher {
+    /// NodeId of the created/edited node.
```

Comment:
If you are referring to a name from documentation, please use Rust syntax for referencing existing names (I dont remember it right now, but its something with `[` and `]`)